### DISCUSSION TITLE
请问下after_clean无法删除build文件是什么原因

### DISCUSSION THREAD

**Discussion by chanchann** [xmake-io/xmake#3245] [Q&A]
workflow的src/xmake.lua中定义 after_clean 脚本，但是 xmake clean 无法删除 build 文件夹， 但是我移动到了 根目录下的 xmake.lua 则能正常删除

请问下导致这样的原因是什么？

(脚本中都能正常删除 _include和_lib, 说明代码是走到这里了的，但是却无法正常删除 build文件夹

修改的 pr 在这里，麻烦大佬指点下，https://github.com/sogou/workflow/pull/1139/files

**Comment by waruqi** [comment]
你放到 workflow target 下 rm build 目录，确实是删了的，但是删了后，之后又 clean 了其他 target ，可能在这个过程中，不知道哪里需要用到 build 目录，导致重新创建了一个空的 build 目录，具体原因还得后面细查，你可以先挪到 root 下。

但你挪到 root 下，每个 target 的 clean 完都会执行 after_clean 去删一遍 build 目录，所以删干净了。。

**Reply by waruqi** [reply]
是内部 c++ modules rule 引入的一些空目录提前创建导致，回头我会改进下

**Reply by waruqi** [reply]
我会在这个 patch 里修复，https://github.com/xmake-io/xmake/pull/3246

**Reply by chanchann** [reply]
非常感谢大佬！
